lib/fluent/plugin/in_tail.rb

```diff
@@ -283,6 +293,28 @@ def have_read_capability?
         @capability.have_capability?(:effective, :dac_override)
     end
 
+    def extended_glob_pattern(path)
+      path.include?('*') || path.include?('?') || /\[.*\]/.match(path)
+    end
+
+    # Curly braces is not supported with default path_delimiter
+    # because the default delimiter of path is ",".
+    # This should be collided for wildcard pattern for curly braces and
+    # be handled as an error on #configure.
+    def use_glob?(path)
+      if @glob_policy == :always
+        # For future extensions, we decided to use `always' term to handle
+        # regular expressions as much as possible.
+        # This is because not using `true' as a returning value
+        # when choosing :always here.
+        extended_glob_pattern(path) || /\{.*,.*\}/.match(path)
```

Comment:
Thanks! Now I can read the intent of the code.

This is fine, but if we reduce comments as much as possible, I would do the following.

```suggestion
        # Intended to be equivalent to `extended_glob_pattern(path) || /\{.*,.*\}/.match(path)`
        return true
```

Would you prefer to keep it as is? If so, I would not object anymore.
Let's merge this!

Thanks for helping us out so far!

Reply:
If so, could we simplify the comments as follows?

```suggestion
        # To reduce the cost of calling `Dir.glob` every time
        extended_glob_pattern(path) || /\{.*,.*\}/.match(path)
```

Reply:
Your suggestion will completely remove the intent of the why we choose _always_ term in the glob_policy option.
Just simplifying is not good option to describe "simply" why we choose this.

Reply:
OK!
We don't need to force simplification, but isn't the following information the most important?

> there is a discussion for how much the calling costs between Dir.glob and regular expressions. I'd say validating with regular expressions might be lower costs to process all the specified paths.

Reply:
Not sure. I don't have a benchmark result for it. 
The most important things should be described there:
* Why we choose _always_ term in the glob_policy option
* Display possible regular expressions there (as a code not in the comment)

Reply:
FYI: I can describe the reason why I wrote the above comment.
_I'd say_ implies that I'm going to say what is not certainly things but I wanted to tell
_might_ implies 50% percent or less possibility. Note that it's not equivaIent to "たぶん" in Japanese. That equivalent is _probably_ not _might_.

Reply:
Thanks! I see.

With just the current comment, I misread the codes as https://github.com/fluent/fluentd/pull/4401#discussion_r1551357082.
So, I'd be glad if we could clarify the intent of the code.

Even if it's only 50%, it would be good to leave it as a comment.
For example, like this.

```suggestion
        # For future extensions, we decided to use `always' term to handle
        # regular expressions as much as possible.
        # It might be costly to call `Dir.glob` every time.
        # This is because not using `true' as a returning value
        # when choosing :always here.
        extended_glob_pattern(path) || /\{.*,.*\}/.match(path)
```

However, I understand the disadvantages of having more comments.
In addition, we can fix those comments later.
So, I think it's fine as it is!

Reply:
Thanks for continuously patient reviewing, @daipom. You rock! :muscle: 💎